src/client/Client.cc

```diff
@@ -7550,6 +7550,21 @@ int Client::_do_setattr(Inode *in, struct ceph_statx *stx, int mask,
   }
 
   if (in->caps_issued_mask(CEPH_CAP_FILE_EXCL)) {
+    if ((mask & CEPH_SETATTR_SIZE) && !(mask & CEPH_SETATTR_KILL_SGUID)) {
+      if ((uint64_t)stx->stx_size < mdsmap->get_max_filesize()) {
+        in->size = in->reported_size = stx->stx_size;
+        ldout(cct,10) << "changing size to " << stx->stx_size << dendl;
+      } else { //too big!
+        ldout(cct,10) << "unable to set size to " << stx->stx_size << ". Too large!" << dendl;
+        return -CEPHFS_EFBIG;
+      }
+      in->cap_dirtier_uid = perms.uid();
+      in->cap_dirtier_gid = perms.gid();
+      in->mark_caps_dirty(CEPH_CAP_FILE_EXCL);
+      mask &= ~(CEPH_SETATTR_SIZE);
+      mask |= CEPH_SETATTR_MTIME;
```

Comment:
We could avoid some code duplication when updating inode (citme, uid,..) with what's done in the block below.